
# Large-scale RNA-Seq Transcriptome Analysis of 4043 Cancers and 548 Normal Tissue Controls across 12 TCGA Cancer Types


Authors (@slack): Sanzida Akhter Anee (@Sanzida), Sk Arif (@arif_shaikh), Mina Zakaria (@mina_zakaria) and Benedict Orile Ajunku (@orile)


## Video Link
## Introduction

Cancer is a disease that results when the abnormal cells divide in an uncontrolled way and spread into other tissues called metastasis, the main reason for the death from the cancer. There are more than 100 types of cancer. The fundamental cause of cancer is mutations in certain genes or epigenetic modifications, and the characterization of alterations in gene expression will advance a large number of new potential diagnostic and therapeutic targets for cancer. 

## Objectives

In this study, genome-wide association analysis of RNA-sequencing data from various TCGA cancer types has been documented. RNA-seq is a new technology of gene expression profiling for identifying transcriptomics-level changes in cancer cells or tissues. The main aim of this study is to analyze gene expression alterations across twelve different cancer types by using RNA-seq data from the Cancer Atlas Genome (TCGA) to identify gene signatures that trigger tumorigenesis and metastasis. 

In this experiment,  transcriptomes of 4043 cancer and 548 normal tissue samples from 21 TCGA cancer types were analyzed and provided gene expression alterations for each cancer type. 

## Summary of key findings

- Identified gene sets that were commonly altered across multiple cancers and gene sets which are specific to certain cancer types.

- Gene signatures (Table 2), sets of genes whose expression levels distinguish between cancer and normal samples (Figure 3).

- Investigated seven cross-cancer gene signatures (Figure 2) that were altered in at least four types of human cancers BLCA, BRCA, LUAD, and LUSC. GO analysis, KEGG analysis, and Pathway Commons analysis indicate that genes in these clusters are enriched in pathways related to cell cycle regulation, which are a common mechanism underlying tumorigenesis. So, this observation supports the hypothesis that abnormalities in cell cycle-associated pathways directly contribute to the progression of cancer. 




- In addition, the results revealed that four gene sets are significantly altered across seven cancer types (Table 2) which indicate that there are common mechanisms underlying the initiation and/or development of human cancers from different organs or different tissues in the same organ.

-  14-gene signatures that classified cancerous and normal samples across seven cancer types and 21 lung cancer-specific gene signatures, containing SFTPA1 and SFTPA2 genes (Figure 4), that helps to distinguish lung cancer samples from other types of cancers. 

- After that, the gene signatures were validated (Figure 5) using non-TCGA datasets that can be used as potential biomarkers for cancer diagnosis and treatment.




## Conclusion

Conclusion: This study emphasizes the importance of analyzing data across multiple cancer types to uncover commonalities in gene expression patterns and identify potential therapeutic targets for cancer.


## Challenges  and future research

The challenge in this study is to make a link between the mutational profiles and differential expression (DE) profiles in cancer. Certain genes (like TP53 and PIK3CA) are frequently mutated in various cancers and significantly contribute to tumor development, but they do not always rank highly in the DE gene list. There are not any specific methods to connect between these two profiles.

So the two key points can be focused on further research:

  - The relationship between mutational and DE profiles
 - How gene expression patterns change in the vicinity of driver mutations

## Workflow
## References

Peng, L., Bian, X. W., Li, D. K., Xu, C., Wang, G. M., Xia, Q. Y., & Xiong, Q. (2015). Large-scale RNA-Seq transcriptome analysis of 4043 cancers and 548 normal tissue controls across 12 TCGA cancer types. Scientific reports, 5(1), 13413.

